terra/executor/resources.py

```diff
@@ -264,25 +306,52 @@ def __enter__(self):
     return self.acquire()
 
   def __exit__(self, exc_type, exc_value, exc_tb):
-    # Does not release, want to reuse resource for life of thread/process
-    pass
+    self.release()
+    return None
 
   def __del__(self):
-    if self._local.resource_id:
-      self.release()
-
-
+    if self.is_locked:
+      logger.warning(f"A {self.name} resource was not released. Cleaning up on delete.")
+      self.release(force=True)
+
+
+# class CachedResource(Resource):
+#   '''
+#   Like the :class:`Resource`, but instead of using normal context managers,
+#   it uses `cached context managers <https://www.python.org/dev/peps/pep-0343/
+#   #caching-context-managers>`_
```

Comment:
will this link render correctly even though it is split with a line break?

Reply:
I also do not think your reading of that section is correct. I do not think that it says it is ok to have a with statement that does not revert its state